//
// This file is part of the WoW++ project.
// 
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 2 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software 
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//
// World of Warcraft, and all World of Warcraft or Warcraft art, images,
// and lore are copyrighted by Blizzard Entertainment, Inc.
// 

syntax = "proto2";
package wowpp.proto;

message SpellEffect {
	required uint32 index = 1;
	optional int32 basepoints = 2;
	optional int32 diesides = 3;
	optional int32 basedice = 4;
	optional float diceperlevel = 5;
	optional float pointsperlevel = 6;
	optional uint32 mechanic = 7;
	optional uint32 targeta = 8;
	optional uint32 targetb = 9;
	optional float radius = 10;
	optional uint32 aura = 11;
	optional int32 amplitude = 12;
	optional float multiplevalue = 13;
	optional uint32 chaintarget = 14;
	optional uint32 itemtype = 15;
	optional int32 miscvaluea = 16;
	optional int32 miscvalueb = 17;
	optional uint32 triggerspell = 18;
	optional uint32 summonunit = 19;
	optional int32 pointspercombopoint = 20;
}

message SpellEntry {
	required uint32 id = 1;
	required string name = 2;
	repeated uint32 attributes = 3;
	repeated SpellEffect effects = 4;
	optional uint64 cooldown = 5;
	optional uint32 casttime = 6;
	optional int32 powertype = 7;
	optional uint32 cost = 8;
	optional uint32 costpct = 9;
	optional int32 maxlevel = 10;
	optional int32 baselevel = 11;
	optional int32 spelllevel = 12;
	optional float speed = 13;
	optional uint32 schoolmask = 14;
	optional uint32 dmgclass = 15;
	optional int32 itemclass = 16;
	optional uint32 itemsubclassmask = 17;
	repeated uint32 skillsonlearnspell = 18;
	optional uint32 facing = 19;
	optional int32 duration = 20;
	optional int32 maxduration = 21;
	optional uint32 interruptflags = 22;
	optional uint32 aurainterruptflags = 23;
	optional float minrange = 24;
	optional float maxrange = 25;
	optional uint32 rangetype = 26;
	optional uint32 targetmap = 27;
	optional float targetx = 28;
	optional float targety = 29;
	optional float targetz = 30;
	optional float targeto = 31;
	optional uint32 maxtargets = 32;
	optional uint32 talentcost = 33;
	optional uint32 procflags = 34;
	optional uint32 procchance = 35;
	optional uint32 proccharges = 36;
}

message Spells {
	repeated SpellEntry spells = 1;
}
